### ISSUE TITLE
How did u build the 6D pose pool?

### ISSUE THREAD

**Issue by swjtuccy** [wadimkehl/ssd-6d#16]
in the paper i see u regress the 2d bbox, then u build a 6d pose pool on the object.

I am confused ,because u don't have the pose and the distance to ur camrea of an object ,how to get the (x,y,z) and the pose pool?

**Comment by fabi92** [comment]
Please refer to section "3.3.1 From 2D bounding box to 6D hypothesis" of the paper.

**Comment by swjtuccy** [comment]
thank u for help.
Do you take different viewpoints from the hemisphere and mark the box manually, then use it for training?

**Comment by swjtuccy** [comment]
> Please refer to section "3.3.1 From 2D bounding box to 6D hypothesis" of the paper.

thank u for help.
Do you take different viewpoints from the hemisphere and mark the box manually, then use it for training?

**Comment by wadimkehl** [comment]
Since we render the objects with their CAD models for the training data, we have access to the metric information. Additionally, with the knowledge of camera intrinsics, we can infer their approximate pose. 
Check out the two functions to understand how it works:

https://github.com/wadimkehl/ssd-6d/blob/b04b2c8035dbbf7383960611f7049171f2429122/rendering/utils.py#L88 

https://github.com/wadimkehl/ssd-6d/blob/b04b2c8035dbbf7383960611f7049171f2429122/rendering/utils.py#L133

**Comment by aditya2592** [comment]
Hi, is it possible to compute viewpoint and in-plane rotation given camera pose and object pose in 6D?

**Comment by wadimkehl** [comment]
Well, it can be done. You could decompose the rotation matrix (in object space) into Euler angles, similar to http://nghiaho.com/?page_id=846, and then compute the polar coordinates for the viewpoint. You would have to figure out, though, which angles correspond to what quantity!

**Comment by aditya2592** [comment]
Thanks for the explanation, so in this case the viewpoint can lie on a spherical space around the object? And how can we get the in-plane rotation once we get the viewpoint ?
